test/testmultiwindow.c

```diff
@@ -0,0 +1,241 @@
+/*
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely.
+*/
+
+/* Program that creates a window at the global position of the mouse */
+
+#define SDL_MAIN_USE_CALLBACKS 1
+#include <SDL3/SDL_main.h>
+#include <SDL3/SDL_test.h>
+#include <SDL3/SDL_test_common.h>
+
+typedef struct {
+  SDL_Window *window;
+  SDL_Renderer *renderer;
+  SDL_Color color;
+} TestWindow;
+
+static void DestroyTestWindow(TestWindow *testWindow)
+{
+  SDL_DestroyRenderer(testWindow->renderer);
+  SDL_DestroyWindow(testWindow->window);
+}
+
+static void RenderTestWindow(TestWindow *testWindow)
+{
+  SDL_Renderer *renderer = testWindow->renderer;
+
+  if (renderer) {
+    SDL_SetRenderDrawColor(renderer,
+      testWindow->color.r,
+      testWindow->color.g,
+      testWindow->color.b,
+      testWindow->color.a);
+    SDL_RenderClear(renderer);
+    SDL_RenderPresent(renderer);
+  }
+}
+
+#define MAX_WINDOWS 8
+
+typedef struct {
+  SDLTest_CommonState *state;
+  TestWindow *testWindows[MAX_WINDOWS];
+  Uint32 lastCreate;
+} TestState;
+
+static void DestroyTestState(TestState *testState)
+{
+  int i;
+  SDLTest_CommonDestroyState(testState->state);
+  for (i = 0; i < MAX_WINDOWS; ++i) {
+    if (testState->testWindows[i]) {
+      DestroyTestWindow(testState->testWindows[i]);
+    }
+  }
+}
+
+static TestWindow *createTestWindowAtMousePosition(SDLTest_CommonState *state)
+{
+  static int windowId = 0;
+
+  TestWindow *testWindow;
+
+  char title[1024];
+  SDL_Rect rect;
+  SDL_Rect bounds;
+  SDL_PropertiesID props;
+  float x, y;
+
+  SDL_GetDisplayUsableBounds(state->displayID, &bounds);
+
+  SDL_snprintf(title, SDL_arraysize(title), "#window%d", ++windowId);
+
+  SDL_GetGlobalMouseState(&x, &y);
+  rect.x = (int)SDL_ceilf(x);
+  rect.y = (int)SDL_ceilf(y);
+
+  rect.w = state->window_w;
+  rect.h = state->window_h;
+  if (state->auto_scale_content) {
+    float scale = SDL_GetDisplayContentScale(state->displayID);
+    rect.w = (int)SDL_ceilf(rect.w * scale);
+    rect.h = (int)SDL_ceilf(rect.h * scale);
+  }
+
+  /* Don't make a window if it wouldn't be visible in the display */
+  if (!SDL_HasRectIntersection(&rect, &bounds)) {
+    return NULL;
+  }
+
+  props = SDL_CreateProperties();
+  SDL_SetStringProperty(props, SDL_PROP_WINDOW_CREATE_TITLE_STRING, title);
+  SDL_SetStringProperty(props, SDL_PROP_WINDOW_CREATE_EMSCRIPTEN_CANVAS_ID, title);
```

Comment:
I think this test needs a custom HTML shell file: https://emscripten.org/docs/compiling/Deploying-Pages.html#build-files-and-custom-shell

Reply:
emscripten support files can be stored in `test/emscripten`.

Reply:
This test is not specific to Emscripten. It works on desktops and also tests multiple things:

- Creating multiple windows
- Unique rendering to each window
- Getting the global mouse position and spawning a window at that position
- Destroying windows

Reply:
That's good, but without HTML items to show windows, the test won't work on emscripten.

Reply:
I've verified this PR with this test on Emscripten. So, it does work.

In this PR, a new canvas element is created in `SDL_CreateWindow` if the canvas ID specified wasn't already present in the DOM.